tools/performance-benchmarking/src/md/0010-content.md

```diff
@@ -0,0 +1,916 @@
+  passed: {}
+  coverage:
+    unit: %
+benchmarks:
+  s2n_init:
+    tags: [s2n, crypto, cbmc]
+    variants:
+        optimized:
+          metrics:
+            runtime: 3
+            passed: true
+            coverage: 100
+          path: ./tmp/benchcomp/runs/abcd1234/optimized/all_cbmc_proofs/s2n/tests/cbmc/proofs/s2n_init
+        release:
+          metrics:
+            runtime: 4
+            passed: true
+            coverage: 100
+          path: ./tmp/benchcomp/runs/abcd1234/release/all_cbmc_proofs/s2n/tests/cbmc/proofs/s2n_init
+  freertos_list_replace:
+    tags: [cbmc, freertos, freertos_kernel]
+    variants:
+      optimized:
+        metrics:
+          # ...
+```
+</div>
+<div class="col col-25">
+
+The output file first declares the various *metrics* that the union of all the parsers emitted.
+In later sections, we will see how Alice writes her own parser or extends an existing one with new metrics.
+The file then maps benchmark names to values for each of the metrics for each of the variants that the benchmark ran under.
+
+</div>
+</div>
+
+## Visualizing the results
+
+Alice now wants to visualize the difference in runtime between the two versions of CBMC, ignoring other metrics for now.
+Alice adds the `visualize` top-level key to her `benchcomp.yaml` file:
+
+<div class="twocolumn">
+<div class="col col-66">
+
+```yaml
+# add to benchcomp.yaml
+
+visualize:
+  - type: dashboard
+    output: comparison.html
+    graphs:
+      - type: pairwise_box_whisker
+        metric: runtime
+        pairs:
+          - [release, optimized]
+```
+
+</div>
+<div class="col col-33">
+
+This says to generate a single graph in a single HTML dashboard visualization.
+The graph compares the runtimes of all benchmarks run under the `release` variant pairwise with the same benchmarks run under the `optimized` variant.
+
+</div>
+</div>
+
+Alice runs the entire suite, together with generating and writing out the visualization, by running `benchcomp` again.
+Alternatively, she can run `benchcomp visualize < result.json`, which loads the result of the run she did in the previous section.
+
+The resulting dashboard looks like this:
+
+<div class="subpage">
+<div class="sidebar">
+<div class="side-header">
+
+Run ID: `abc123`
+
+`2023-01-01T18:42:54`
+
+[JSON version](/) of this dashboard
+
+</div>
+<div class="tags-bar">
+<div class="tags-header">
+
+**Filter dashboards by tags**
+
+</div>
+<div class="tags-container">
+
+* [cbmc](/) <span class="n_proofs">(833)</span>
+  * [s2n](/) <span class="n_proofs">(128)</span>
+  * [freertos](/) <span class="n_proofs">(547)</span>
+  * [e-sdk](/) <span class="n_proofs">(49)</span>
+  * [uses-function-contracts](/) <span class="n_proofs">(49)</span>
+
+</div>
+</div>
+</div>
+<div class="central-container">
+<div class="central-view">
+
+
+~include tmp/box_whisker/cbmc.html
+
+</div>
+</div>
+</div>
+
+Benchmarks whose runtime did not change between the two versions have a 'runtime increase' of 1.0.
```

Comment:
Does that mean that if I see s2n(1.0) I get no change in runtime? How precise is that?